static/js/services/search.js

```diff
@@ -80,25 +108,27 @@ var _ = require('underscore'),
 
             var result;
             if (extensions.displayLastVisitedDate) {
-              var lastVisitedDatePromise = getLastVisitedDates(searchResults);
-
-              result = $q.all({
-                dataRecords: dataRecordsPromise,
-                lastVisitedDates: lastVisitedDatePromise
-              }).then(function(r) {
-                r.lastVisitedDates.forEach(function(dateResult) {
-                  var relevantDataRecord = r.dataRecords.find(function(dataRecord) {
-                    return dataRecord._id === dateResult.key;
+              var lastVisitedDatePromise = getLastVisitedDates(searchResults, extensions.visitCountSettings);
+
+              result = $q
+                .all([ dataRecordsPromise, lastVisitedDatePromise ])
```

Comment:
any reason for refactoring out passing a map to `$q.all`? IMO it's more readable, because you're not relying on an array matching up, and you get to actually name things.

Reply:
I refactored it because the `Q` we used in Karma doesn't support mapping and I wanted to make the code testable. `Promise.all` doesn't support mapping either.